src/app/common/hooks/analytics/transactions-analytics.hooks.ts

```diff
@@ -111,3 +118,27 @@ export function useTrackChangedTransactions(
   );
   return result;
 }
+
+function useIsFirstDeposit(stxBalance: AccountStxBalanceBigNumber | undefined): boolean {
+  const currentNetwork = useCurrentNetworkState();
+  const hadZeroStxBalance = useAnalyticsHadZeroStx()[currentNetwork.chainId];
+  const hasZeroStx = stxBalance?.total_received.isEqualTo(0) || false;
+  useEffect(() => {
+    if (hasZeroStx || hadZeroStxBalance) {
+      store.dispatch(
+        analyticsActions.hadZeroStx({ network: currentNetwork.chainId, hadZeroStx: hasZeroStx })
+      );
+    }
+  }, [hasZeroStx, hadZeroStxBalance, currentNetwork.chainId]);
+  return hadZeroStxBalance && !hasZeroStx;
+}
+
+export function useTrackFirstDeposit() {
+  const analytics = useAnalytics();
+  const { data: balances } = useCurrentAccountUnanchoredBalances();
+  const firstDeposit = useIsFirstDeposit(balances?.stx);
+  useEffect(() => {
+    if (!firstDeposit || !balances) return;
+    void analytics.track('first_deposit', { type: 'stx', amount: balances.stx.balance });
+  });
```

Comment:
Same here, we almost never need to use a `useEffect` without a dependency array. This could be the cause of render perf issues, as this check is running on every render.

Reply:
https://maxrozen.com/learn-useeffect-dependency-array-react-hooks